src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,38 @@
 package mate.academy;
 
+import java.util.Map;
+import java.util.Random;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
+    private static final String USER_DETAILS = "Details for ";
+    private static final Random random = new Random();
+
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> getUserData(userId), executor);
+    }
+
+    private UserData getUserData(String userId) {
+        try {
+            Thread.sleep(random.nextInt(1000));
+        } catch (InterruptedException ex) {
+            Thread.currentThread().interrupt();
+        }
+        UserData cachedUserData = cache.get(userId);
```

Comment:
Access to the shared mutable state 'cache' is not synchronized here. While ConcurrentHashMap handles some level of concurrency, if there's a possibility of complex operations on the map, consider using proper synchronization mechanisms to ensure thread safety.